soc/ambiq/apollo4x/power.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2024 Ambiq LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <soc.h>
+
+#include <zephyr/drivers/interrupt_controller/gic.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/pm/pm.h>
+#include <zephyr/init.h>
+
+/* ambiq-sdk includes */
+#include <am_mcu_apollo.h>
+
+LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+
+void pm_state_set(enum pm_state state, uint8_t substate_id)
+{
+	ARG_UNUSED(substate_id);
+
+	__disable_irq();
+	__set_BASEPRI(0);
+
+	switch (state) {
+	case PM_STATE_SUSPEND_TO_IDLE: {
+		/* Put ARM core to normal sleep. */
+		am_hal_sysctrl_sleep(AM_HAL_SYSCTRL_SLEEP_NORMAL);
+		break;
+	}
+	case PM_STATE_SUSPEND_TO_RAM: {
+		/* Put ARM core to deep sleep. */
+		/* Cotex-m: power down, register value preserve.*/
+		/* Cache: power down*/
+		/* Flash: power down*/
+		/* Sram: retention*/
+		am_hal_sysctrl_sleep(AM_HAL_SYSCTRL_SLEEP_DEEP);
+		break;
+	}
+	default: {
+		LOG_DBG("Unsupported power state %u", state);
+		break;
+	}
+	}
+}
+
+void pm_state_exit_post_ops(enum pm_state state, uint8_t substate_id)
+{
+	ARG_UNUSED(substate_id);
+
+	switch (state) {
+	case PM_STATE_SUSPEND_TO_IDLE: {
+		/* Nothing is needed after soc is wake up.*/
+		break;
+	}
+	case PM_STATE_SUSPEND_TO_RAM: {
+		/* Flash, cache, sram automatically switch to active state on wake up,
+		 * Nothing needed for software.
+		 */
+		break;
+	}
+	default: {
+		LOG_DBG("Unsupported power substate-id %u", state);
+		break;
+	}
+	}
```

Comment:
please make compiler's parser life easier by removing these unnecessary 15 lines that do nothing.

Reply:
I will put these comments in the doxygen of this function but I would like the comments to remain in the file for customer's information.